From 7ae06009bd622afa7f6966aa0dac7a41c14cab64 Mon Sep 17 00:00:00 2001
From: Jan Petrous <jan.petrous@nxp.com>
Date: Tue, 21 Mar 2023 14:58:24 +0100
Subject: [PATCH 1/3] s32g: hwconfig fixup: Update PFE EMAC property name

The PFE property name "nxp,pfeng-emac-id" was changed
to "nxp,pfeng-linked-phyif".

Issue: ALB-9925
Upstream-Status: Pending

Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hwconfig_fixups.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
index 712f1bfe9c..702307229f 100644
--- a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
+++ b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright 2022 NXP
+ * Copyright 2022,2023 NXP
  */
 #include <common.h>
 #include <malloc.h>
@@ -42,6 +42,8 @@
 
 #define MAX_PROP_ALLOC			(500)
 
+#define MAX_PFE_EMAC_PHYIF_ID		2
+
 struct dts_node {
 	union {
 		struct {
@@ -1141,10 +1143,13 @@ static int set_xpcs_config(struct dts_node *root, u32 serdes_id, u32 xpcs_id)
 	 * If the property is missing, then we have another type of interface (e.g. GMAC)
 	 * which we do not check if is SGMII (TBD).
 	 */
-	ret = node_get_prop_u32(&node, "nxp,pfeng-emac-id", &emac_id);
+	ret = node_get_prop_u32(&node, "nxp,pfeng-linked-phyif", &emac_id);
 	if (ret) {
 		pr_warn("Interface does not have ID\n");
 		return -ENODATA;
+	} else if (emac_id > MAX_PFE_EMAC_PHYIF_ID) {
+		pr_warn("Interface has no EMAC\n");
+		return -ENODATA;
 	} else if (pfeng_cfg_emac_get_interface(emac_id) != PHY_INTERFACE_MODE_SGMII) {
 		pr_warn("PFE EMAC %d is not SGMII\n", emac_id);
 		return -ENODATA;
-- 
2.25.1

